src/ui/toolbar/GPSIndicator.qml

```diff
@@ -94,17 +94,36 @@ Item {
         anchors.left:           gpsIcon.right
 
         QGCLabel {
-            anchors.horizontalCenter:   hdopValue.horizontalCenter
-            visible:                    _activeVehicle && !isNaN(_activeVehicle.gps.hdop.value)
+            anchors.horizontalCenter:   gpsLock.horizontalCenter
+            visible:                    _activeVehicle && _activeVehicle.gps.count.valueString !== ""
             color:                      qgcPal.buttonText
             text:                       _activeVehicle ? _activeVehicle.gps.count.valueString : ""
         }
 
         QGCLabel {
-            id:         hdopValue
-            visible:    _activeVehicle && !isNaN(_activeVehicle.gps.hdop.value)
-            color:      qgcPal.buttonText
-            text:       _activeVehicle ? _activeVehicle.gps.hdop.value.toFixed(1) : ""
+            id:         gpsLock
+            visible:    _activeVehicle && _activeVehicle.gps.lock.enumStringValue !== ""
+            color:      getLockColor()
+            text:       getLockText()
+
+            function getLockColor() {
+                // Make the text orange if we don't have RTK lock
+                if (!_activeVehicle) return qgcPal.buttonText
+                const lockEnum = _activeVehicle.gps.lock.enumIndex
+                if (lockEnum === 3 || lockEnum === 4) return "orange"
```

Comment:
Is there a awy to connect this to an Enum, e.g. the Mavlink definitions?

Reply:
Im not sure I understood what you meant here, did you mean to connect create an enum for the color of the text?

Reply:
I'm thinking about 3 and 4 being hard to understand, if there is some enum/constant available with their names that we don't have to define ourselves, that's perferrable. Perhaps gps.lock type is such an enum?

Reply:
I was not able to find any enum like this unfortunately, but that would have been better. I added a comment about enumIndex = 4 aswell, to explain all relevant values